token-lending/program/src/processor.rs

```diff
@@ -1185,13 +1185,19 @@ fn process_borrow_obligation_liquidity(
         return Err(LendingError::BorrowTooSmall.into());
     }
 
+    let cumulative_borrow_rate_wads = borrow_reserve.liquidity.cumulative_borrow_rate_wads;
+
     borrow_reserve.liquidity.borrow(borrow_amount)?;
     borrow_reserve.last_update.mark_stale();
     Reserve::pack(borrow_reserve, &mut borrow_reserve_info.data.borrow_mut())?;
 
-    obligation
-        .find_or_add_liquidity_to_borrows(*borrow_reserve_info.key)?
-        .borrow(borrow_amount)?;
+    let obligation_liquidity =
+        obligation.find_or_add_liquidity_to_borrows(*borrow_reserve_info.key)?;
```

Comment:
How about passing `cumulative_borrow_rate_wads` in to `Obligation::find_or_add_liquidity_to_borrows` so that `ObligationLiquidity::new` is always called with an up-to-date `cumulative_borrow_rate_wads`?